src/core/IronPython/Hosting/PythonCommandLine.cs

```diff
@@ -309,6 +312,59 @@ private void InitializeModules() {
             }
 
             PythonContext.SetHostVariables(prefix ?? "", executable, null);
+
+
+            // --- Local functions -------
+
+            static bool FindRunner(string prefix, string name, string assembly, out string runner) {
+                runner = null;
+#if NET
+                while (prefix != null) {
+                    runner = Path.Combine(prefix, name);
+                    if (File.Exists(runner)) {
+                        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) || IsExecutable(runner)) {
+                            break;
+                        }
+                    }
+                    prefix = Path.GetDirectoryName(prefix);
+                }
+                if (runner != null && Path.GetExtension(assembly).Equals(".dll", StringComparison.OrdinalIgnoreCase)) {
+                    // make sure that the runner refers to this DLL
+                    var relativeAssemblyPath = assembly.Substring(prefix.Length + 1); // skip over the path separator
```

Comment:
Hmm, I guess `prefix` here could be `null` if it doesn't find anything?

Reply:
Indeed, the test is wrong. Thanks!